### ISSUE TITLE
startRemoteDisplayStream does not work if VideoStreamingParameters is null

### ISSUE THREAD

**Issue by shiniwat** [smartdevicelink/sdl_android#666]
### Bug Report
startRemoteDisplayStream does not work if VideoStreamingParameter is null.
According to the API description, VideoStreamingParameter is explained as:
"streaming parameters to be used when streaming. If null is sent in, the default/optimized options will be used. If you are unsure about what parameters to be used it is best to just send null and let the system determine what works best for the currently connected module."

##### Reproduction Steps
1. Launch SDL app that does video projection. 
2. Make sure using virtual display encoder and specify VideoStreamingParameter to null when using startRemoteDisplayStream.

##### Expected Behavior
Video projection should start with appropriate video streaming parameter.

##### Observed Behavior
onProtocolSessionStartedNACKed() gets called, and failed to start session.

##### OS & Version Information
* Android Version: observed on Android 6.0.1, but the issue should be OS version neutral.
* SDL Android Version: 4.4.0
* Testing Against: Our dev board for video projection.

##### Test Case, Sample Code, and / or Example App
I'm ready for PR once this issue is recognized as the bug.

**Comment by brettywhite** [comment]
Hi @shiniwat 

I am looking this over and having a hard time replicating it. I was able to pass in null and start the video session. For reference, I was using this branch of our test video app: https://github.com/livio/sdl_video_streaming_android_sample/tree/test/new_apis  with the 4.4 SDK and using a Ford TDK as the head unit and USB/AOA as the transport. Can you try our example app and see if it works for you? 

In `SdlService.java` in the `startVideoStream` function, you will want to make sure you use the enhanced function to take advantage of the new call. Additionally, it might be worth making sure your development board supports the default settings as defined in `VideoStreamingParameters.java` 

Please let me know if any of that helps you out. 

Thank you!

**Comment by shiniwat** [comment]
Hi @brettywhite 

Thanks for the comment. As we investigated sdl_video_streaming_android_sample you specified above, and the issue needs additional precondition for reproduce -- you'll get this repro'ed if you do getCapability(SystemCapabilityType.VIDEO_STREAMING) prior to startRemoteDisplayStream.

Please try adding the following code to SdlService.java (around line#478), something like below:
```
        if(!notification.getHmiLevel().equals(HMILevel.HMI_NONE)
                && firstNonHmiNone){
            sendCommands();
            firstNonHmiNone = false;
        }else{
            //We have HMI_NONE
            if(notification.getFirstRun()){
                uploadImages();
+              proxy.getCapability(SystemCapabilityType.VIDEO_STREAMING);
            }
        }

```
I believe you'll run into Service NACKed when you start session.

Thanks.

**Comment by brettywhite** [comment]
So if you are using the new API in 4.4 using `startRemoteDisplayStream` It will check your protocol version and then call getSystemCapability appropriately. My guess is that you are using an older version that doesn't support `VIDEO_STREAMING`. 

You can see the logic here: https://github.com/smartdevicelink/sdl_android/blob/master/sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java#L4048-L4089

It will fall back if its the older version. Again, this is a guess regarding your system by me. Try it out without that call in HMI status and let me know how it goes.

**Comment by shiniwat** [comment]
Hi @brettywhite 
We're using the new 4.4 API, and understand startRemoteDisplayStream automatically gets VideoStreamingCapability. 
For some reason, our app needs to know the HU's VideoStreamingCapability prior to actually startRemoteDisplayStream. After we get the VideoStreamingCapability from HU, we have some options to specify video streaming parameters, and one of the options is to rely on HU's VideoStreamingCapability (== null parameter). In that case, the problem happens as I explained here.

**Comment by brettywhite** [comment]
I just gave your code a try in `sdlService` with a head unit that that call would error on that call (protocol version 4). Following, I was able to start video streaming successfully using the new API (it fell back to the proper defaults). Are you making sure that you have `HMI_FULL` before starting the stream? It is possible that your code may be trying to start the stream in a level that isn't `HMI_FULL`, causing it to fail. Let me know! Thanks

**Comment by shiniwat** [comment]
The issue is specific to WiProVerion >= 5. 
I added SDL Core version (4.4.1) to original report (in OS & Version information).

**Comment by brettywhite** [comment]
I tried it on 5 as well. When are you trying to start the video stream? It's important to note that the order in which things are called is important. The `VideoStreamingCapability` can be called in HMI_NONE (before the app is displayed) and more likely than not you will have its callback before the app is displayed. If you try and start the video service prior to HMI_FULL, it will fail. Not being able to see how you are doing that, it is hard to help from here. A good test would be to comment out wherever you are starting video from right now and call it when the HMI_level becomes HMI_FULL then you know it will be at the right time and see if that works.

**Comment by shiniwat** [comment]
As I mentioned before, the issue reproduces if
0) Recipient (SDL Core) is 4.4.1 (WiProVersion is 5)
1) proxy.getCapability(SystemCapabilityType.VIDEO_STREAMING) when HMI_NONE.
2) After getting HMI_FULL, startRemoteDisplayStream with specifying null to VideoStreamingParameter.

Then you'll get Service NACK'ed.
I believe the time to call getCapability and startRemoteDisplayStream are valid.

Well, more specifically, the issue can be solved if I change SystemCapabilityManager.java to the commit below:
https://github.com/shiniwat/sdl_android-1/commit/c89e56be5afacd07cb1ccc3a2219f5c72f5e71bc

I suspect this is the bug in SystemCapabilityManager.

**Comment by joeygrover** [comment]
closed with #669
